File: Api/src/PWPA.CallLogging.BackEnd.ApplicationCore/Models/Call.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace PWPA.CallLogging.BackEnd.ApplicationCore.Models;

public class Call
{
    [BsonId]
    [BsonRepresentation(BsonType.ObjectId)]
    public string? Id { get; set; }

    [BsonRequired]
    public string? CallerName { get; set; }

    [BsonRequired]
    public string? Address { get; set; }

    [BsonRequired]
    public string? Description { get; set; }
}

File: FrontEnd/src/models/Call.ts
export interface Call {
    callerName: String;
    address: String;
    description: String;
}

export class CallFormModel implements Call {

    constructor(callerName : String, address: String, description: String) {
        this.callerName = callerName,
        this.address = address,
        this.description = description
    }

    callerName : String = '';
    address : String = '';
    description : String = '';
}